dirs/dirs.go

```diff
@@ -190,6 +198,73 @@ func init() {
 	SetRootDir(root)
 }
 
+// SnapHomeDirs returns a slice of the home directories currently in use.
+func SnapHomeDirs() []string {
+	snapHomeDirsMu.Lock()
+	defer snapHomeDirsMu.Unlock()
+	dirs := make([]string, len(snapHomeDirs))
+	copy(dirs, snapHomeDirs)
+	// Should never be true since SetSnapHomeDirs is ran on init and on SetRootDir calls.
+	// Useful for unit tests.
+	if len(dirs) == 0 {
+		return []string{filepath.Join(GlobalRootDir, "/home")}
+	}
+	return dirs
+}
+
+// SetSnapHomeDirs sets SnapHomeDirs to the user defined values and appends /home if needed.
+// Homedir must be a comma separated list of the user defined home directories.
+// If homedirs is empty, SnapHomeDirs will be a slice of length 1 containing "/home".
+// Also generates the data directory globbing expressions for each user.
+// Ran at startup by configstate.Init, returns the split outsnap home directories. Use DataHomeGlobs to retrieve globs.
+func SetSnapHomeDirs(homedirs string) []string {
+	snapHomeDirsMu.Lock()
+	defer snapHomeDirsMu.Unlock()
+
+	//clear old values
+	snapHomeDirs = nil
+	snapDataHomeGlob = nil
+	hiddenSnapDataHomeGlob = nil
+
+	// Do not set the root directory as home unless explicitly specified with "."
+	if homedirs != "" {
+		snapHomeDirs = strings.Split(homedirs, ",")
+		for i := range snapHomeDirs {
+			// clean the path
+			snapHomeDirs[i] = filepath.Clean(snapHomeDirs[i])
+			globalRootDir := GlobalRootDir
+			// Avoid false positives with HasPrefix
+			if globalRootDir != "/" {
+				globalRootDir = globalRootDir + "/"
+			}
+			if !strings.HasPrefix(snapHomeDirs[i], globalRootDir) {
+				snapHomeDirs[i] = filepath.Join(GlobalRootDir, snapHomeDirs[i])
+			}
+			// Generate data directory globbing expressions for each user.
+			snapDataHomeGlob = append(snapDataHomeGlob, snapHomeDirs[i]+"/*/"+UserHomeSnapDir)
+			hiddenSnapDataHomeGlob = append(hiddenSnapDataHomeGlob, snapHomeDirs[i]+"/*/"+HiddenSnapDataHomeDir)
+		}
+	}
+
+	// Make sure /home is part of the list.
+	hasHome := false
+	for _, e := range snapHomeDirs {
+		if e == filepath.Join(GlobalRootDir, "/home") {
```

Comment:
We should probably compute this once, and not in a loop.

Reply:
I had this originally implemented using `slices.Contains` however as @ZeyadYasser noted in #https://github.com/snapcore/snapd/pull/13145#discussion_r1354848966 the package containing it is an experimental one so I had to revert to using a loop. 
Maybe there is a package in golang that can do this that I am not aware of?

Reply:
How about something simple, like this?

```go
globalHome := filepath.Join(GlobalRootDir, "/home")
for _, e := range snapHomeDirs {
   if e == globalHome {
      ...
   }
   ...
}